remotes/docker/config/hosts.go

```diff
@@ -100,8 +100,16 @@ func ConfigureHosts(ctx context.Context, options HostOptions) docker.RegistryHos
 				hosts[len(hosts)-1].host = "registry-1.docker.io"
 			} else if docker.IsLocalhost(host) {
 				hosts[len(hosts)-1].host = host
-				if options.DefaultScheme == "" || options.DefaultScheme == "http" {
-					hosts[len(hosts)-1].scheme = "http"
+				if options.DefaultScheme == "" {
+					_, port, _ := net.SplitHostPort(host)
+					if port == "" || port == "443" {
+						//if !strings.ContainsAny(host, ":") || strings.HasSuffix(host, ":443") {
```

Comment:
Is this line intended?

Reply:
Updated, was just there for checking the unit test 😁